assets/src/js/admin/paypal-commerce/index.js

```diff
@@ -103,51 +113,158 @@ window.addEventListener( 'DOMContentLoaded', function() {
                     },
                 };
                 const paypalErrorQuickHelp = document.getElementById( 'give-paypal-onboarding-trouble-notice' );
+                const getPartnerLinkAjaxRequest = async () => {
+                    // Request partner obboarding link.
+                    fetch(ajaxurl + `?action=give_paypal_commerce_get_partner_url&countryCode=${countryCode}&mode=${mode}&accountType=${connectionAccountType ?? 'EXPRESS_CHECKOUT'}`)
+                        .then(response => response.json())
+                        .then(function (res) {
+                            if (true === res.success) {
+                                const payPalLink = document.querySelector('[data-paypal-button]');
+
+                                // Dynamically set callback function name.
+                                payPalLink.setAttribute(
+                                    'data-paypal-onboard-complete',
+                                    'live' === mode
+                                        ? 'giveLivePayPalOnBoardedCallback'
+                                        : 'giveSandboxPayPalOnBoardedCallback'
+                                );
+
+                                // Set PayPal button link (Partener link).
+                                payPalLink.href = `${res.data.partnerLink}&displayMode=minibrowser`;
+
+                                payPalLink.click();
+                            }
```

Comment:
The point of using `async` is to clean up all the promises with all the `.then().then().then()` stuff with callback hell.
```js
const response = await fetch();
const data = await response.json();

// do the things
```

Reply:
fixed: https://github.com/impress-org/givewp/pull/6867/commits/b26e5fdc431592754346d0f145c836af0155ed13